test/fail_compilation/retscope.d

```diff
@@ -267,3 +267,34 @@ void escape5() @safe
     pp = &p; // error
 }
 
+/***********************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(287): Error: escaping reference to local variable b
+---
+*/
+
+@safe int* foo6()(int* arg)
+{
+    return arg;
+}
+
+int* escape6() @safe
+{
+    int b;
+    return foo6(&b);
+}
+
+/***************************************************/
+
+struct S7
+{
+    int[10] a;
+    int[3] abc(int i)
```

Comment:
> `scope` is not transitive, so the indirection is not covered by it.

So this should not error ? If so, please add the test case as well.
Also, `abc` is still missing `@safe`.